arelle/FileSource.py

```diff
@@ -167,10 +167,11 @@ def __init__(self, url: str, cntlr: Cntlr | None = None, checkIfXmlIsEis: bool =
         self.url = str(url)  # allow either string or FileNamedStringIO
         self.baseIsHttp = isHttpUrl(self.url)
         self.cntlr = cntlr
-        self.type = self.url.lower()[-7:]
-        self.isTarGz = self.type == ".tar.gz"
-        if not self.isTarGz:
-            self.type = self.type[3:]
+        self.isTarGz = self.url.endswith(".tar.gz")
```

Comment:
This conflicts with and is already handled in master. It looks like the base branch for this PR is a month or two out of date (and has merge conflicts). Can you update your working branch with the latest commits from master?